compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -278,9 +278,13 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                             OperandValue::Immediate(v) => v,
                             OperandValue::Pair(_, _) => todo!(),
                         };
-                        // FIXME: find the real vtable!
-                        let trait_ref = get_trait_ref(bx.tcx(), mir_cast_ty);
-                        let vtable = get_vtable(bx.cx(), source.ty(self.mir, bx.tcx()), Some(trait_ref));
+                        let trait_ref =
```

Comment:
This doesn't fix anything, it just seems more correct to me when I was scanning the code. That `get_trait_ref` function walking through to look for the dyn's principal type is very... suspicious, and we don't need to use it.

Reply:
Just to make sure I understand the terminology, the principal type is the `Trait` part of `dyn* Trait`? And so `get_trait_ref` is kind of overkill for what we need, or more general, when we can just ask for the part we care about directly?

Reply:
Correct. There's at most one principal trait in a trait object, and that's the thing that the vtable corresponds to.

I'm not certain that `get_trait_ref` is even correct -- it's only used in one place for a seemingly niche use case, but I don't want to touch it.

Reply:
Also this no longer ICEs (no principal):

```rust
#![feature(async_fn_in_traits)]

fn main() {
    let x: dyn* Send = 1usize as _;
}
```